tools/base/fpgaconf/bitstream-tools.h

```diff
@@ -168,58 +171,55 @@ fpga_result check_bitstream_guid(const uint8_t *bitstream)
 	return FPGA_OK;
 }
 
-fpga_result get_fpga_interface_id(fpga_token token, uint64_t *id_l,
-								uint64_t *id_h)
+fpga_result get_fpga_interface_id(fpga_token token,
+						uint64_t *id_l,
+						uint64_t *id_h)
 {
-	fpga_result result = FPGA_OK;
-	char buf_l[INTFC_ID_LOW_LEN + 1] = { 0 };
-	char buf_h[INTFC_ID_HIGH_LEN + 1] = { 0 };
-	uint8_t buffer[INTFC_ID_LOW_LEN + INTFC_ID_HIGH_LEN] = { 0 };
-
-	fpga_object pr_object;
+	fpga_result result         = FPGA_OK;
+	fpga_result resval         = FPGA_OK;
+	fpga_properties filter     = NULL;
+	fpga_guid guid;
 	errno_t e;
 
-	result = fpgaTokenGetObject(token, PR_INTERFACE_ID, &pr_object, 0);
+	result = fpgaGetProperties(token, &filter);
 	if (result != FPGA_OK) {
-		OPAE_ERR("Failed to get Token Object \n");
-		return result;
+		OPAE_ERR("Failed to get Token Properties Object \n");
+		goto out;
 	}
-
-	result = fpgaObjectRead(pr_object, buffer, 0,
-		INTFC_ID_LOW_LEN + INTFC_ID_HIGH_LEN, 0);
+	result = fpgaPropertiesGetGUID(filter, &guid);
 	if (result != FPGA_OK) {
-		OPAE_ERR("Failed to Read Object \n");
-		return result;
+		OPAE_ERR("Failed to get PR guid \n");
+		goto out_destroy;
 	}
 
-	// PR Inteface Id h
-	memset_s(buf_h, sizeof(buf_h), 0);
-	e = strncpy_s(buf_h, sizeof(buf_h), (char *) buffer, INTFC_ID_HIGH_LEN);
+	e = memcpy_s(id_h, sizeof(id_h),
+		guid, sizeof(uint64_t));
 	if (EOK != e) {
-		OPAE_ERR("strncpy_s failed (buf_l)\n");
-		return FPGA_EXCEPTION;
+		OPAE_ERR("memcpy_s failed");
+		goto out_destroy;
 	}
 
-	*id_h = strtoull(buf_h, NULL, 16);
+	*id_h = bswap_64(*id_h);
```

Comment:
Wondering if the byte swapping should be done here or internally by the properties layer?  Or is this for compatibility with the bitstream metadata format?

Reply:
OPAE properties layer doesn't swap bytes

Reply:
Understood.. I'm wondering whether it should be done at the properties level?

Reply:
fixed in tools